src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets

```diff
@@ -5,12 +5,14 @@
     <IlcExportUnmanagedEntrypoints>true</IlcExportUnmanagedEntrypoints><!-- For JSExports -->
   </PropertyGroup>
   <ItemGroup>
-    <EmccExportedFunction Include="_free" />
-    <EmccExportedFunction Include="_htons" />
-    <EmccExportedFunction Include="_malloc" />
-    <EmccExportedFunction Include="_memalign" />
-    <EmccExportedFunction Include="_memset" />
-    <EmccExportedFunction Include="_ntohs" />
+    <EmccNativeExportedFunction Include="free" />
+    <EmccNativeExportedFunction Include="htons" />
+    <EmccNativeExportedFunction Include="main" />
+    <EmccNativeExportedFunction Include="malloc" />
+    <EmccNativeExportedFunction Include="memalign" />
+    <EmccNativeExportedFunction Include="memset" />
+    <EmccNativeExportedFunction Include="ntohs" />
+    <EmccExportedFunction Include="_%(EmccNativeExportedFunction)" />
```

Comment:
Here's a snag I have not considered: `EmccExportedFunction` is a public API documented as being directly fed into Emscripten, i. e. you're expected to mangle it beforehand by adding the underscore to C symbols, but _not_ symbols defined in JS libraries. So it's hard to feed that into ILC as-is: no matter what you do, JS symbols will come out mangled on the other side.

In light of this, I think we have to keep the exports file rewriting for now (see e. g. https://github.com/emscripten-core/emscripten/issues/8380 for what the future might look like).

We could add another option to ILC that says "don't mangle symbols", but it doesn't seem worth it at this point in time.

BTW, there is a bit of code in the file under `Condition="'$(ExportsFile)' == ''` that could be deleted because the exports file is now always created so all of the exports can be routed through it.

Reply:
Aye aye